### ISSUE TITLE
Connection failed (no connection to the remote host). Internal error: 1. Error details: Code: 10060

### ISSUE THREAD

**Issue by ds-combine-ai** [Azure-Samples/cognitive-services-speech-sdk#1397]
**Describe the bug**
Hi, I'm having issue connecting to the service.  I'm running the sample c# quickstart code given here, which uses a wav file https://docs.microsoft.com/en-us/azure/cognitive-services/speech-service/get-started-speech-to-text?tabs=windowsinstall&pivots=programming-language-csharp#recognize-from-file.

Console output:
```
RECOGNIZED: Text=
CANCELED: Reason=Error
CANCELED: ErrorCode=ConnectionFailure
CANCELED: ErrorDetails=Connection failed (no connection to the remote host). Internal error: 1. Error details: Code: 10060. SessionId: 0e23a3cf83f24e4d897398eef70874a1
CANCELED: Did you update the speech key and location/region info?
```

log snippet:
```
[3289]: 106275ms SPX_TRACE_ERROR: AZ_LOG_ERROR:  tlsio_schannel.c:1177 xio_open failed
[3289]: 106275ms SPX_TRACE_ERROR:  trace_message.cpp:207 Error: File:D:\a\_work\1\s\external\azure-c-shared-utility\adapters\tlsio_schannel.c Func:tlsio_schannel_open Line:1177
[3289]: 106275ms SPX_TRACE_ERROR: AZ_LOG_ERROR:  uws_client.c:1781 Opening the underlying IO failed
[3289]: 106275ms SPX_TRACE_ERROR:  trace_message.cpp:207 Error: File:D:\a\_work\1\s\external\azure-c-shared-utility\src\uws_client.c Func:uws_client_open_async Line:1781
[170989]: 106275ms SPX_DBG_TRACE_VERBOSE:  handle_table.h:106 CSpxHandleTable::TrackHandle p=0x00000262ABD023D0
[3289]: 106275ms SPX_TRACE_ERROR:  web_socket.cpp:533 uws_client_open_async failed with result 1783
[170989]: 106275ms SPX_DBG_TRACE_VERBOSE:  handle_table.h:116 CSpxHandleTable::TrackHandle class=ISpxRecognitionResult, h=0x00000262ABD023D0, p=0x00000262ABD023D0, tot=1
[3289]: 106275ms SPX_TRACE_ERROR: AZ_LOG_ERROR:  web_socket.h:73 Failed to change state for WebSocket 00000262ABB6FD30. From: 1, To: 0, Current: 0
[3289]: 106275ms SPX_TRACE_ERROR:  trace_message.cpp:207 Error: File:D:\a\_work\1\s\source\core\networking\web_socket.h Func:?ChangeState@CSpxWebSocket@Impl@Speech@CognitiveServices@Microsoft@@IEAAXW4WebSocketState@2345@0@Z Line:73
[170989]: 106275ms SPX_DBG_TRACE_VERBOSE:  handle_table.h:175 CSpxHandleTable::StopTracking(h) h=0x00000262ABB63A90
[3289]: 106275ms SPX_TRACE_ERROR:  web_socket.cpp:769 Failed to open transport
```

Seems to be an issue with the websocket connecting, but I can't figure out why.  I double checked the subscription key and region (see below) and I'm sure these are ok.  I've included the full logs below.

**To Reproduce**

1. Get source code from https://docs.microsoft.com/en-us/azure/cognitive-services/speech-service/get-started-speech-to-text?tabs=windowsinstall&pivots=programming-language-csharp#recognize-from-file
2. Install Speech SDK using Nuget
3. Download sample wav file from https://github.com/Azure-Samples/cognitive-services-speech-sdk/blob/master/samples/csharp/sharedcontent/console/whatstheweatherlike.wav
3. Run sample

**Expected behavior**
The sample wav file to be converted into speech and to appear in the console.

**Version of the Cognitive Services Speech SDK**
1.20.0

**Platform, Operating System, and Programming Language**
 - OS: Windows Server 2019 Datacenter, version 1809, OS Build 17763.2452
 - Hardware - x64
 - Programming language: C#, running in Visual Studio Community 2019 16.11.9
 - Browser - N/A

**Additional context**
- full log is here: [speechlog.txt](https://github.com/Azure-Samples/cognitive-services-speech-sdk/files/7968628/speechlog.txt)
- Visual Studio Solution details: 
```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp3.1</TargetFramework>
    <Platforms>AnyCPU;x64</Platforms>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="Microsoft.CognitiveServices.Speech" Version="1.20.0" />
  </ItemGroup>
</Project>
```
- I've checked that web sockets feature is installed under Server Manager
- I've checked that web sockets appear in IIS and are enabled in the configuration editor
- I've run a docker container for the Speech CLI on my mac (following this guide https://docs.microsoft.com/en-us/azure/cognitive-services/speech-service/spx-basics?tabs=dockerinstall%2Cterminal), using the same subscription key, region and wav file.  This works perfectly first time, so I am sure that the key/region are correct.

**Comment by dargilco** [comment]
@ds-combine-ai this is Darren from the Speech SDK team. Thank you for opening this issue and providing such detailed information, including the log file. That's very helpful. Let me look at the log and consult with my team before getting back to you.
Darren.

**Comment by ds-combine-ai** [comment]
Thanks @dargilco :)

**Comment by dargilco** [comment]
@ds-combine-ai can you open this web site on your Windows Server to check if it supports web sockets? http://websocketstest.com/

Also, can you run the Java Script Speech SDK browser demo on your Windows Server and see if that works? https://azure.microsoft.com/services/cognitive-services/speech-to-text/#features

**Comment by ds-combine-ai** [comment]
Thanks @dargilco for getting back to me. I've just tried both those tests.  The first one worked ok...but the second one failed.  Here are the screenshots:
<img width="748" alt="Screenshot 2022-02-02 at 12 51 38" src="https://user-images.githubusercontent.com/73110276/152091417-0106eef3-b4a3-40c4-a625-a8fbdac277c4.png">

<img width="1333" alt="Screenshot 2022-02-02 at 13 00 57" src="https://user-images.githubusercontent.com/73110276/152091425-7afdc800-5491-435b-ad82-fd25061cca31.png">

I have a feeling it's going to be some small tiny bit of config somewhere that is tripping it up...do you have any ideas?

**Comment by dargilco** [comment]
@ds-combine-ai thank you for the screen shots. It does look like something in your Server configuration. To check access via REST APis to our Speech Service, please go to this link https://eastus.stt.speech.microsoft.com/api/v1.0/languages/translation/from?format=detailed on the Service and let me know if you get a response (a long JSON string).

**Comment by ds-combine-ai** [comment]
Thanks @dargilco - I got the same issue:
<img width="776" alt="Screenshot 2022-02-03 at 15 43 11" src="https://user-images.githubusercontent.com/73110276/152294009-6b00c337-ee77-4485-ab7a-3e296328a9e0.png">

I tried accessing the same link from my Mac and it worked, no problem - I got the long JSON string as you said.

**Comment by ds-combine-ai** [comment]
Hey @dargilco, I've been wracking my brains and trying every kind of configuration options I can but still can't get any response for that link.  There isn't any chance it's being blocked on the MS end is there?  I've tried turning off my firewall completely, but still no luck:

<img width="701" alt="Screenshot 2022-02-08 at 12 41 12" src="https://user-images.githubusercontent.com/73110276/153124296-71e8985e-7fe4-4e0d-af49-52efd6ea072b.png">

I also tried running the diagnostics options presented by Edge in the previous comment:

<img width="568" alt="Screenshot 2022-02-08 at 13 06 58" src="https://user-images.githubusercontent.com/73110276/153124374-27eda3ef-2054-45a9-99cf-1c1885a12089.png">

I can PM you the reports if that is any help?  Thanks for your help

**Comment by ds-combine-ai** [comment]
Hi @dargilco.  I finally figured this out, so just letting you know what the problem was.  After much searching and reconfiguring, it turned out this was the thread that pointed me in the right direction https://docs.microsoft.com/en-us/answers/questions/452532/not-able-to-access-the-cognitive-service-urlfrom-i.html

I had tried a few times to create a Speech Service in my Azure, and I think during one of these times I was following the instructions on this screen to configure security to `Selected networks`.  

<img width="783" alt="Screenshot 2022-02-16 at 16 44 37" src="https://user-images.githubusercontent.com/73110276/154219231-3fb55300-9f14-4225-8cba-01280131f7e8.png">

After selecting the subnet, I followed the instructions given here:

<img width="683" alt="Screenshot 2022-02-16 at 16 44 54" src="https://user-images.githubusercontent.com/73110276/154219300-d78d01a7-3012-4f5e-8dae-95777eaf4b00.png">

This is what it looks like in the subnet configuration screen:

<img width="1428" alt="Screenshot 2022-02-16 at 14 11 44" src="https://user-images.githubusercontent.com/73110276/154219569-f68c57f2-835c-4fed-986c-09ffe7c9b900.png">

So for some reason this was the thing blocking any communications with the servers.  Once I found this option and turned it off, it all started working again.

I'm not sure if this is a SDK issue or not, should I close or leave open for you?
